### ISSUE TITLE
CADDYPATH env variable not respected

### ISSUE THREAD

**Issue by PaluMacil** [mholt/caddy#2515]
## 1. Which version of Caddy are you using (`caddy -version`)?

Caddy 0.11.5 (+80dfb8b Mon Mar 04 19:50:49 UTC 2019) (unofficial)
Though it says unofficial, I downloaded from the releases page via `wget https://github.com/mholt/caddy/releases/download/v0.11.5/caddy_v0.11.5_linux_amd64.tar.gz`

## 2. What are you trying to do?

I am trying to serve my site with tls, but I don't want to save to my root users' home folder. I thought I could do this by setting caddypath. I'm doing it in Systemd, but I can recreate the issue with a very simple couple commands...

```
env CADDYPATH=/etc/ssl/caddy
/usr/local/bin/caddy -log=stdout -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
```

If I run this as user www-data, it fails with `decoding certificate metadata: unexpected end of JSON input` which seems to indicate a permission issue. When I instead use root, I see that it ignores CADDYPATH. It shows output that indicates it is trying to use the home folder of root: `Activating privacy features... 2019/03/11 05:54:57 [INFO][FileStorage:/root/.caddy] Started certificate maintenance routine`

If www-data is also ignoring CADDYPATH, it might be failing on not having access to root's home.

## 3. What is your Caddyfile?

Almost nothing...

```
danwolf.net {
        proxy / localhost:3035
}
```

## 4. How did you run Caddy (give the full command and describe the execution environment)?

ssh into environment and run:

```
env CADDYPATH=/etc/ssl/caddy
/usr/local/bin/caddy -log=stdout -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
```

## 5. Please paste any relevant HTTP request(s) here.

NA

## 6. What did you expect to see?

I expected the ssl certs to be cached in `/etc/ssl/caddy`

## 7. What did you see instead (give full error messages and/or log)?

It ignored CADDYPATH and used the root users home dir, adding a .caddy folder there.

## 8. Why is this a bug, and how do you think this should be fixed?

CADDYPATH is pretty basic functionality. I assume it worked until the last release due to the prevalence in docs. If I can't use it, I can't seem to use a non-root user to run caddy.

## 9. What are you doing to work around the problem in the meantime?

I have shut down my website and will wait for a response before returning to Nginx.

## 10. Please link to any related issues, pull requests, and/or discussion.

I didn't see any related complaints, but I did see that #2465 made a similar complaint. It could be related.

## Bonus: What do you use Caddy for? Why did you choose Caddy?

I use IIS mostly, so for my personal website I wanted to branch out and try something friendly.

## More info...

Ideally I'd like to run as www-data and I'm basically using the systemd init file suggested in the download of caddy shown below. It fails with  `decoding certificate metadata: unexpected end of JSON input` whether run via systemd or via `sudo -E -u www-data /usr/local/bin/caddy -log=stdout -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp` even though I can run it as root.

```
[Unit]
Description=Caddy HTTP/2 web server
Documentation=https://caddyserver.com/docs
After=network-online.target
Wants=network-online.target systemd-networkd-wait-online.service

[Service]
Restart=on-abnormal

; User and group the process will run as.
User=www-data
Group=www-data

; Letsencrypt-issued certificates will be written to this directory.
Environment=CADDYPATH=/etc/ssl/caddy

; Always set "-root" to something safe in case it gets forgotten in the Caddyfile.
ExecStart=/usr/local/bin/caddy -log stdout -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
ExecReload=/bin/kill -USR1 $MAINPID

; Use graceful shutdown with a reasonable timeout
KillMode=mixed
KillSignal=SIGQUIT
TimeoutStopSec=5s
```

**Comment by mholt** [comment]
I am unable to reproduce this; can you please just run Caddy like this:

```bash
$ CADDYPATH=/etc/ssl/caddy caddy -log stderr -conf=/ets/caddy/Caddyfile -root=/var/tmp -ca "https://acme-staging-v02.api.letsencrypt.org/directory"
```

It should prompt you for an email address and then proceed to obtain certificates through their staging endpoint, so it is safe for testing.

If you need to run as root, please do so for that test.

**Comment by PaluMacil** [comment]
@mholt I very much appreciate your rapid response. I believe this is a non issue, but for the sake of others after me, I will reconstruct what I did:

1. I created the proper folders and applied the proper permissions for www-data to be the user running Caddy.
2. I tested with root and no config path specified, which made the config in my home dir.
3. I then must have tested with the `/etc/ssl/caddy` path for config but root before testing as www-data, which created internal folders such as users and sites that were owned by root.
4. Finally, I tested as www-data and since there were some folders owned by root, it failed, but I checked the folder ownership of the outtermost `/etc/ssl/caddy` folder, which looked correct, which led me to think it was recursively correct still. It wasn't.

I still think there might be a bug because when I set `/etc/ssl/caddy` path for config and was testing with root (shown in my post), the console output seemed to indicate that I was using root's home to store config. However, it's probably an issue of working too late at night and making another user error with my setting of the environmental variable to set the config, but I can't seem to identify how I would have done that. I wasn't setting my env variable on the same line as the caddy command before you posted above, so it's possible I misunderstand how setting env variables work, but I didn't think so.

If I find there is a logging error, I'll of course open a different issue.
